### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Gmanlove**
In this pull request ♻️

- [ ] Solve the N+1 problem when fetching in this URL (users/:id/posts) 👍

- [ ] Setup Capybara to write integration tests 👍

- [ ] write all required tests

Screenshot 2023-06-19 at 14 04 20
Screenshot 2023-06-19 at 14 01 46

- [ ] This is the Screeshot of the passing test

Screenshot 2023-06-23 at 07 10 30

### CODE REVIEW

File: spec/features/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    # Test code goes here
+  end
+
+  it 'displays the number of posts each user has written' do
+    # Test code goes here
+  end
+
+  it 'redirects to user show page when clicking on a user' do
+    # Test code goes here
+  end
```
Comment (Owusu-Desmond) [inline spec/features/user_index_spec.rb]: - [ ] Good to have this user index test file. :+1:  But please from the project requirement, you should write a test for the following :point_down: But now you have no test code in your `it` block to test the following:blush: 
  
  -   Tests that asserts you can see the username of all other users.
  -   Tests that asserts you can see the profile picture for each user.
  -   Tests that asserts you can see the number of posts each user has written.
  -   Tests that redirects to the user's show page when you click on a user.

File: spec/features/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  it 'displays the number of likes the post has' do
+    # Test code goes here
+  end
+
+  it 'displays the post body' do
+    # Test code goes here
+  end
+
+  it 'displays the username of each commentor' do
+    # Test code goes here
```
Comment (Owusu-Desmond) [inline spec/features/post_show_spec.rb]: - [ ] Good to have this post-show test file. :+1:  But please from the project requirement, you should write a test for the following :point_down: But now you have no test code in your `it` block to test the following:blush: 

  * Tests that asserts you can see the post's title.
  * Tests that asserts you can see who wrote the post.
  * Tests that asserts you can see how many comments it has.
  * Tests that asserts you can see how many likes it has.
  * Tests that asserts you can see the post body.
  * Tests that asserts you can see the username of each commentor.
  * Tests that asserts you can see the comment each commentor left.

File: spec/features/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  it 'redirects to post show page when clicking on a user post' do
+    # Test code goes here
+  end
+
+  it 'redirects to user post index page when clicking to see all posts' do
+    # Test code goes here
+  end
+end
```
Comment (Owusu-Desmond) [inline spec/features/user_show_spec.rb]: - [ ] Good to have this user show test file. :+1:  But please from the project requirement, you should write a test for the following :point_down: But now you have no test code in your `it` block to test the following:blush: 
  
  * Tests that asserts you can see the user's profile picture.
  * Tests that asserts you can see the user's username.
  * Tests that asserts you can see the number of posts the user has written.
  * Tests that asserts you can see the user's bio.
  * Tests that asserts you can see the user's first 3 posts.
  * Tests that asserts you can see a button that lets me view all of a user's posts.
  * Tests that redirects to the post's show page when you click on a user's post.
  * Tests that redirects to the user's post's index page when you click on a **see all posts**.

File: spec/features/user_post_index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    # Test code goes here
+  end
+
+  it 'displays the number of comments a post has' do
+    # Test code goes here
+  end
+
+  it 'displays the number of likes a post has' do
+    # Test code goes here
+  end
```
Comment (Owusu-Desmond) [inline spec/features/user_post_index_spec.rb]: - [ ] Good to have this user post index test file. :+1:  But please from the project requirement, you should write a test for the following :point_down: But now you have no test code in your `it` block to test the following:blush: 
  
  * Tests that asserts you can see the user's profile picture.
  * Tests that asserts you can see the user's username.
  * Tests that asserts you can see the number of posts the user has written.
  * Tests that asserts you can see a post's title.
  * Tests that asserts you can see some of the post's body.
  * Tests that asserts you can see the first comments on a post.
  * Tests that asserts you can see how many comments a post has.
  * Tests that asserts you can see how many likes a post has.
  * Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
  * Tests that redirects to the post's show page when you click on a post.

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @Gmanlove and @belalgamal46 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
📌 Your linters are passing✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Your readme file is okay ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb
Comment (KDlamini) [inline spec/features/post_index_spec.rb:1]: Great job adding all the required specs 👏🏾💯

However, all your tests fail on my end.

image

- If the failing tests are due to an error on my system and they all pass on your end. Kindly reply with a screenshot of your passing tests. Thank you 🙏🏾

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @Gmanlove, @belalgamal46 🙋‍♀️,  

You have done an incredible job implementing some of the changes requested by the previous reviewer. 👏🏾👏🏾👏🏾  
However, there are still some changes that aren't addressed yet. You are almost there! 💪🏾 

Highlights 🌞

✔️ Good PR title and description
✔️ Capybara and Rspec are used for tests
✔️ Tests for User and Posts pages are implemented

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb
Comment (Gmanlove) [reply spec/features/post_index_spec.rb:1]: You should have contacted me before returning for request changes....  All the test are passing at my end....  Now I have only one review left.....

Comment (Gmanlove) [reply spec/features/post_index_spec.rb:1]: This is the Screenshot of my passing test. thanks 

- [ ] 
Screenshot 2023-06-23 at 07 10 30

**Review by brytebee — APPROVED** [review]
Hi, @Gmanlove 👋,

##

### Highlights💡
- Great updating the setup and seed.✔️
- Well-written tests.✔️
- All tests passed.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
Practice more about 
- the seeding of a database.
- use of capybara

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**